src/main/java/UseCase/RegisterInteractor.java

```diff
@@ -0,0 +1,41 @@
+package UseCase;
+
+import Entities.User;
+import OutputBoundary.loginOutputBoundary;
+import OutputData.loginOutputData;
+import InputData.RegisterInputData;
+import InputBoundary.RegisterInputBoundary;
+import Database.userAccessInterface;
+import Database.userDsData;
+import Entities.userFactory;
+
+public class RegisterInteractor implements RegisterInputBoundary{
+    private final loginOutputBoundary registerPresenter;
+    private final userFactory userFactory;
+
+    private final userAccessInterface allUsers;
+
+    public RegisterInteractor(loginOutputBoundary registerPresenter, userAccessInterface allUsers,
```

Comment:
Similar to my previous comment why are you passing an interface into the constructor?